src/classes/RD_RecurringDonations_TEST2.cls

```diff
@@ -835,11 +835,24 @@ private with sharing class RD_RecurringDonations_TEST2 {
         npe03__Recurring_Donation_Field__c = 'Name'
         );
         
+        test.startTest();
+        
         ApexPages.StandardController scc = new ApexPages.StandardController(r1);
         RD_AddDonationsBTN_CTRL rdadb = new RD_AddDonationsBTN_CTRL(scc);
         system.assertEquals(r1.id, rdadb.sc.getId());
-        rdadb.buttonClick();    
-    }    
+        rdadb.buttonClick();
+       
+        //Read Only user should be denied permission to add donation
```

Comment:
Added an extra bit to an existing test rather than write a new test.   Didn't think this use case was worth the overhead of another test (since our builds are already slow).  Not great BDD test technique but easy enough to break out into it's own test if we think that is clearer.